LEGO1/mxcollection.h

```diff
@@ -0,0 +1,25 @@
+#ifndef MXCOLLECTION_H
+#define MXCOLLECTION_H
+
+#include "mxcore.h"
+
+template <class T>
+class MxCollection : public MxCore {
+public:
+	MxCollection()
+	{
+		m_count = 0;
+		m_customDestructor = Destroy;
+	}
+
+	virtual ~MxCollection() {}
+
+	static void Destroy(T){};
+	virtual MxBool Compare(T, T) { return 0; }
```

Comment:
Are we sure about the `MxBool` return type? Can it actually be distinguished from a `(unsigned) char` in the assembly? I'm wondering since negative values can be returned from all the Compare implementations we have so far.

Reply:
oh duh. yeah, the one from `MxVariableTable` is just a wrapper around `strcmp`. I'll fix it